app/controllers/roles_controller.rb

```diff
@@ -0,0 +1,64 @@
+class RolesController < ApplicationController
+  before_action :set_role, only: %i[show new_user new_role remove_role]
+
+  def index
+    @roles = Role.where(role_type: 'staff_role')
+  end
+
+  def new
+    @role = Role.new
+    @role.role_privileges.build
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def create
+    @role = Role.new(role_params)
+    params[:role_privileges].each_value do |role|
+      @privilege = @role.role_privileges.new(privilege_params(role))
+    end
+    if @role.save
+      redirect_to roles_path, notice: "Created a new
+                              #{@role.name} role!".squish
+    else
+      redirect_to new_role_path, alert: @role.errors.full_messages
+    end
+  end
+  # rubocop:enable Metrics/AbcSize
```

Comment:
Please remove these rubocop exclusions and refactor, or leave a comment to explain why this issue should be excluded.